### PR TITLE
update build configuration for 0.6+

### PR DESCRIPTION
**Pull request by appleboy**
Signed-off-by: Bo-Yi Wu <appleboy.tw@gmail.com>

### CODE REVIEW

**Comment by tboerger** [comment]
Should this be the new generalized format? Has @bradrydzewski said anything to that?

**Comment by appleboy** [comment]
See the changes: https://github.com/drone-plugins/drone-s3/commit/3351e6323bc87699bff8cc309c57e7bef779ceb7

**Comment by tboerger** [comment]
Commented on a specific package that have been needed: https://github.com/drone-plugins/drone-s3/commit/3351e6323bc87699bff8cc309c57e7bef779ceb7#commitcomment-23841564

**Comment by appleboy** [comment]
@tboerger Good point. waiting for @bradrydzewski response

**Comment by tboerger** [comment]
Maybe @bradrydzewski can create a drone/scratch repo, there we can add the alpine ca-certificates.crt and mime.types and use it as base image for the plugins. That's even better because it's in the control of Drone and can get updated by us without issues. Looks like the https://github.com/CenturyLinkLabs/ca-certs-base-image image is already 2 years old, so the ca-certificates.crt can already be changed a lot.

**Comment by appleboy** [comment]
agree with @bradrydzewski maybe we can create base image naming as `plugin/base` first or using multiple stage build. Which one do you prefer? @tboerger 

```dockerfile
FROM plugin/base
ADD release/linux/amd64/drone-s3-cache /bin/
ENTRYPOINT ["/bin/drone-s3-cache"]
```

**Comment by appleboy** [comment]
@bradrydzewski @tboerger I already add mime types file.

**Comment by tboerger** [comment]
I would personally prefer a base image to reduce the noise on all the plugin builds, but in the end the decision is up to @bradrydzewski. Maybe add images as plugins/base:scratch and plugins/base:alpine? Quite simple base images of course, but only a single place to build them instead of multi stage builds over and over again :)

**Comment by bradrydzewski** [comment]
Looks good. I think can probably omit mime.types from this plugin since it is uploading tarballs for internal use only.

**Comment by appleboy** [comment]
@bradrydzewski can you add `docker_username` and `docker_password` on drone-plugins org in drone?

https://beta.drone.io/drone-plugins/drone-hg/14

**Comment by tboerger** [comment]
> @bradrydzewski can you add docker_username and docker_password on drone-plugins org in drone?
> 
> https://beta.drone.io/drone-plugins/drone-hg/14

Not really related to this issue but I have added the secrets.

**Comment by appleboy** [comment]
@bradrydzewski I removed the `mime.types` file.

File: .drone.yml [hunk 1]
```diff
@@ -3,10 +3,47 @@ workspace:
...
     commands:
+      - go vet
       - go test -cover -coverprofile=coverage.out
-      - go build -ldflags "-s -w -X main.revision=$(git rev-parse HEAD)" -a
+
+  build:
+    image: golang:1.9
+    commands: sh .drone.sh
+    when:
+      branch: master
```
Comment (tboerger) [inline .drone.yml]: IMHO it should always build the binaries just to make sure it can be built for all architectures

Comment (appleboy) [reply .drone.yml]: @tboerger Fixed.

File: .drone.yml [hunk 2]
```diff
@@ -3,10 +3,49 @@ workspace:
...
+  publish_linux_amd64:
+    image: plugins/docker
+    group: tags
+    repo: plugins/s3-cache
+    tags: [ "latest", "1.1", "1" ]
+    secrets: [ docker_username, docker_password ]
+    when:
+      branch: master
+      event: push
+      status: success
```
Comment (tboerger) [inline .drone.yml]: I don't think that you need that, this should be the default

**Comment by donny-dont** [comment]
What about a windows build?

**Comment by tboerger** [comment]
Windows should also work without issues

File: .drone.yml (hunk 2 above)
Comment (appleboy) [reply .drone.yml]: Done. @tboerger

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,13 @@
-# Docker image for the Drone build runner
-#
-#     CGO_ENABLED=0 go build -a -tags netgo
-#     docker build --rm=true -t plugins/drone-cache .
+FROM centurylink/ca-certs
```
Comment (tboerger) [inline Dockerfile]: Also stop using this image?

**Comment by tboerger** [comment]
Looks like you have applied most of the stuff I have also introduced. So I'm fine with that if we found a consensus

**Comment by appleboy** [comment]
@tboerger Yes,

File: Dockerfile (hunk 3 above)
Comment (appleboy) [reply Dockerfile]: why stop using this image?

Comment (tboerger) [reply Dockerfile]: Because it's not maintained for more than a year, drone will also use a different image.

Comment (appleboy) [reply Dockerfile]: I see you build the image using docker multiple stages mode.

**Comment by appleboy** [comment]
@tboerger Great. Thank you.

**Pull request merged**